src/hooks.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useRef } from "react"
+
+export function useDidMountEffect(fn, deps) {
+  const renderedOnce = useRef(false)
```

Comment:
Should this be `useState` or are you trying to store this boolean separately from the state?

Reply:
it needs to not be `useState` because updating state triggers a re-render. `useRef` is the current 'escape hatch' that lets you just have a little state container you can write a value to without doing anything with the react lifecycle. It's not the best pattern, but the only real way to do what I'm trying to do here.

Reply:
it's sort of the equivalent of putting a value directly on `this` if you're writing a react component with an ES6 class